src/animalShelter/AnimalShelter.ts

```diff
@@ -16,93 +29,98 @@ export default class AnimalShelter {
 
   private animals: Animal[] = [];
 
-  private constructor() { }
-
+  private constructor() {}
 
   static getInstance() {
     if (!AnimalShelter.instance) {
       AnimalShelter.instance = new AnimalShelter();
     }
     return AnimalShelter.instance;
   }
+  static resetInstance() {
```

Comment:
that is not good. Test code should not change / add functionality to execution code. Please find another way to test the animalShelter

Reply:
If you need to reset the animals and limits in the shelter somehow you can create a method to do so, but creating a new instance is not a good idea in my opinion.

Reply:
@kdimitrov03  i thought this can be an instance method. yo ualready have access to the instance, just make a normal (not static method) that does "this._animals = []; this._mammalLimit = .....